File: frontend/src/objects/order/OrderPriceSnapshot.ts
import type { ProductId } from '@/objects/shared/ids'

export interface OrderPriceSnapshotItem {
  productId: ProductId
  name: string
  unitPrice: number
  quantity: number
  lineAmount: number
}

export interface OrderPriceSnapshot {
  items: OrderPriceSnapshotItem[]
  productOriginalAmount: number
  merchantDiscountAmount: number
  voucherDiscountAmount: number
  platformDiscountAmount: number
  deliveryFeeAmount: number
  discountAmount: number
  payableAmount: number
  merchantReceivableAmount: number
  appliedPromotionTitles?: string[]
  usedVoucherTitle?: string | null
}

File: backend/src/order/objects/OrderPriceSnapshot.scala
package delivery.order.objects

import delivery.shared.objects.ProductId

final case class OrderPriceSnapshotItem(
    productId: ProductId,
    name: String,
    unitPrice: Double,
    quantity: Int,
    lineAmount: Double
)

final case class OrderPriceSnapshot(
    items: List[OrderPriceSnapshotItem],
    productOriginalAmount: Double,
    merchantDiscountAmount: Double,
    voucherDiscountAmount: Double,
    platformDiscountAmount: Double,
    deliveryFeeAmount: Double,
    discountAmount: Double,
    payableAmount: Double,
    merchantReceivableAmount: Double,
    appliedPromotionTitles: List[String] = Nil,
    usedVoucherTitle: Option[String] = None
)
